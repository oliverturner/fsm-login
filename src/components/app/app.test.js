import React from "react";
import ReactDOM from "react-dom";
import { testStateMachine } from "react-automata";

import App from ".";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const fixtures = {
  initialData: {
    user: {},
    error: "You must supply a name"
  },
  loading: {
    SUCCESS: {
      user: "oliverturner"
    },
    FAILURE: {
      error: "You must supply a name"
    }
  }
};

it("works with a stateMachine", () => {
  testStateMachine(App, fixtures);
});
