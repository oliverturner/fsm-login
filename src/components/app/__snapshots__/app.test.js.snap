// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <h2>
    Log in to your account
  </h2>
  <label
    htmlFor="user"
  />
  <input
    className=""
    id="user"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <button
    disabled={false}
  >
    Login
  </button>
</form>
`;

exports[`loading 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <h2>
    Log in to your account
  </h2>
  <label
    htmlFor="user"
  />
  <input
    className=""
    id="user"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <button
    disabled={true}
  >
    Logging in...
  </button>
</form>
`;

exports[`loggedIn 1`] = `
<div>
  <h1>
    Hello 
  </h1>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;

exports[`loggedOut 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <h2>
    Log in to your account
  </h2>
  <label
    htmlFor="user"
  />
  <input
    className=""
    id="user"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <button
    disabled={undefined}
  >
    Login
  </button>
</form>
`;
